src/sentry/static/sentry/app/utils/apm.jsx

```diff
@@ -1,141 +1,24 @@
-    return;
-  }
-
-  clearTimeout(flushTransactionTimeout);
-  wasInterrupted = true;
-}
-
-export function finishTransaction(delay = TRANSACTION_TIMEOUT) {
-  if (flushTransactionTimeout || (hasActiveRenders() || hasActiveRequests())) {
-    interruptFlush();
-  }
-
-  flushTransactionTimeout = setTimeout(() => {
-    if (currentTransactionSpan) {
-      currentTransactionSpan.finish();
-      currentTransactionSpan = null;
-      firstPageLoad = false;
-    }
-  }, delay);
-}
-
-/**
- * These `start-` functions attempt to track the state of "actions".
- *
- * They interrupt the transaction flush (which times out), and
- * requires the related `finish-` function to be called.
- */
-export function startRequest(id) {
-  requests.add(id);
-  interruptFlush();
-}
-export function startRender(id) {
-  renders.add(id);
-  interruptFlush();
-}
-
-/**
- * These `finish-` functions clean up the "active" state of an ongoing "action".
- * If there are no other "actions" and we have interrupted a flush, we should
- * finish the transaction
- */
-export function finishRequest(id) {
-  requests.delete(id);
-  // TODO(apm): Is this necessary? flush should be interrupted already from start()
-  interruptFlush();
-
-  if (wasInterrupted && !hasActiveRenders() && !hasActiveRequests()) {
-    finishTransaction(1);
-  }
-}
-export function finishRender(id) {
-  renders.delete(id);
-  interruptFlush();
-
-  if (wasInterrupted && !hasActiveRenders() && !hasActiveRequests()) {
-    finishTransaction(1);
-  }
-}
+import {TransactionActivity} from '@sentry/integrations';
 
 /**
  * Sets the transaction name
  */
 export function setTransactionName(name) {
-  Sentry.configureScope(scope => {
-    const span = scope.getSpan();
-
-    if (!span) {
-      return;
-    }
-
-    span.transaction = firstPageLoad ? `PageLoad: ${name}` : name;
-    scope.setTag('ui.route', name);
-  });
+  TransactionActivity.updateTransactionName(`${name}`);
+  Sentry.setTag('ui.route', name);
 }
 
 /**
- * This is called only when our application is initialized. Creates a root span
- * and creates a router listener to create a new root span as user navigates.
+ * This is called only when our application is initialized. Creates a transaction
+ * and creates a router listener to create a new transaction as user navigates.
  */
 export function startApm() {
-  Sentry.configureScope(scope => {
-    currentTransactionSpan = Sentry.startSpan({
-      op: 'pageload',
-      sampled: true,
-    });
-    scope.setSpan(currentTransactionSpan);
-    scope.setTag('ui.nav', 'pageload');
+  // `${window.location.href}` will be used a temp transaction name
+  // Internally once our <App> is mounted and the Router is initalized
+  // we call `setTransactionName` to update the full URL to the route name
+  TransactionActivity.startIdleTransaction(`${window.location.href}`, {
+    op: 'pageload',
+    sampled: true,
   });
-  startTransaction();
-  Router.browserHistory.listen(() => startTransaction());
```

Comment:
Why are we getting rid of this completely?

Reply:
This lives now in the integration!
It will berefined, it's still rough around the edges.